daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilContentHandler.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.processors
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream, OutputStream}
+
+import org.apache.daffodil.infoset.SAXInfosetInputter
+import org.xml.sax.{Attributes, ContentHandler, Locator}
+
+class DaffodilContentHandler(dp: DataProcessor, output: OutputStream) extends ContentHandler {
+  private val inputter = new SAXInfosetInputter(new ByteArrayInputStream(output.asInstanceOf[ByteArrayOutputStream].toByteArray))
+  private var _locator: Locator = null
+
+  override def setDocumentLocator(locator: Locator): Unit = {
+    _locator = locator
+  }
+
+  override def startDocument(): Unit = {
+    output.write("<?xml version=\"1.0\" encoding=\"UTF-8\" ?>".getBytes)
```

Comment:
I knwo you probably haven't put much though into this since this is all unparsing, but just note that this is all about unparsing. The OutputStream here is for daffodil to write unparsed data to, not to output the infoset to.